modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-values-factory/src/main/java/com/liferay/dynamic/data/mapping/form/values/factory/internal/DDMFormValuesFactoryImpl.java

```diff
@@ -137,18 +144,30 @@ private void _checkDDMFormFieldParameterNames(
 				ddmFormFieldParameterNames.add(
 					defaultDDMFormFieldParameterName);
 
-				_checkDDMFormFieldParameterNames(
-					ddmFormField.getNestedDDMFormFields(), StringPool.BLANK,
-					ddmFormFieldParameterNames);
+				if (!fullDDMFormFields.containsKey(ddmFormField.getName())) {
+					fullDDMFormFields.put(ddmFormField.getName(), ddmFormField);
+
+					_checkDDMFormFieldParameterNames(
+						ddmFormField.getNestedDDMFormFields(), StringPool.BLANK,
+						ddmFormFieldParameterNames, fullDDMFormFields);
+				}
 			}
 
 			for (String filteredDDMFormFieldParameterName :
 					filteredDDMFormFieldParameterNames) {
 
-				_checkDDMFormFieldParameterNames(
-					ddmFormField.getNestedDDMFormFields(),
-					filteredDDMFormFieldParameterName,
-					ddmFormFieldParameterNames);
+				if (!fullDDMFormFields.containsKey(ddmFormField.getName())) {
+					fullDDMFormFields.put(ddmFormField.getName(), ddmFormField);
+
+					_checkDDMFormFieldParameterNames(
+						ddmFormField.getNestedDDMFormFields(),
+						filteredDDMFormFieldParameterName,
+						ddmFormFieldParameterNames, fullDDMFormFields);
+				}
```

Comment:
@linolaoj this code won't be executed if we add the field in `fullDDMFormFields` on line 148, right?

So, do you guarantee that there is no difference between calling 
```
_checkDDMFormFieldParameterNames(
		ddmFormField.getNestedDDMFormFields(), StringPool.BLANK,
		ddmFormFieldParameterNames, fullDDMFormFields);
```
(line 150) or 

```
_checkDDMFormFieldParameterNames(
		ddmFormField.getNestedDDMFormFields(),
		filteredDDMFormFieldParameterName,
	   ddmFormFieldParameterNames, fullDDMFormFields); 
```
(line 162) even if we are passing `StringPool.BLANK` as `parentDDMFormFieldParameterName` in the first case and `filteredDDMFormFieldParameterName` as `parentDDMFormFieldParameterName` in the second case?

I want to make sure that we really don't need to run this second code if the first one is executed.

Thanks!